AmplifyPlugins/Core/AWSPluginsCore/Model/Support/ModelField+GraphQL.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+
+/// Extension that adds GraphQL specific utilities to `ModelField`.
+extension ModelField {
+
+    /// The GraphQL name of the field.
+    var graphQLName: String {
+        if isAssociationOwner, case let .belongsTo(_, targetName) = association {
+            return targetName ?? name.pascalCased() + "Id"
```

Comment:
Enhancement for later: We should see if we can centralize name generation of these various documents, fields, input types, etc. There's a lot of string manipulation scattered throughout the graphql support code that is hard to remember and refer to later.

Reply:
Added over in "API Plugin Improvement" Issue to track this https://github.com/aws-amplify/amplify-ios/issues/292